### ISSUE TITLE
H100 softmax perf

### ISSUE THREAD

**Issue by kevinstephano** [NVIDIA/Fuser#322]
Repro:

```
import torch
from nvfuser import FusionDefinition, DataType

inputs = [torch.randn(64, 16, 256, 256, device='cuda'),]

def nvfuser_fusion_id0(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(symbolic_sizes=[-1, -1, -1, -1], contiguity=[True, True, True, True], dtype=DataType.Float, is_cpu=False)
    T1 = fd.ops.max(T0, axes=[3], keepdim=False, dtype=DataType.Null)
    T2 = fd.ops.broadcast_in_dim(T1, output_shape=[64, 16, 256, 1], broadcast_dims=[0, 1, 2])
    T3 = fd.ops.broadcast_in_dim(T2, output_shape=[64, 16, 256, 256], broadcast_dims=[0, 1, 2, 3])
    T4 = fd.ops.sub(T0, T3)
    T5 = fd.ops.exp(T4)
    T6 = fd.ops.sum(T5, axes=[3], keepdim=False, dtype=DataType.Null)
    T7 = fd.ops.broadcast_in_dim(T6, output_shape=[64, 16, 256, 1], broadcast_dims=[0, 1, 2])
    T8 = fd.ops.broadcast_in_dim(T7, output_shape=[64, 16, 256, 256], broadcast_dims=[0, 1, 2, 3])
    T9 = fd.ops.div(T5, T8)
    fd.add_output(T9)

with FusionDefinition() as fd:
    nvfuser_fusion_id0(fd)

for _ in range(5):
    out = fd.execute(inputs)
```

Kernel time: `1.163ms`

```
 1551225562        1163307     239  262144     1     1    64     1     1       24         0.000         0.000                                                     NVIDIA H100 PCIe (0)    1     7  CudaCodeGen::kernel1(CudaCodeGen::Tensor<float, (int)4, (int)4>, CudaCodeGen::Tensor<float, (int)4,…
 1552389893        1163691     250  262144     1     1    64     1     1       24         0.000         0.000                                                     NVIDIA H100 PCIe (0)    1     7  CudaCodeGen::kernel1(CudaCodeGen::Tensor<float, (int)4, (int)4>, CudaCodeGen::Tensor<float, (int)4,…
```

**Comment by liqiangxl** [comment]
use CU_FUNC_ATTRIBUTE_PREFERRED_SHARED_MEMORY_CARVEOUT can modify Shared Memory Configuration Size , and kernel can achieve 100% theoretical occupancy ratio. However, the Achieved Occupancy is only 11%.

**Comment by naoyam** [comment]
With #373:

```
$ PYTORCH_NVFUSER_DUMP=dump_eff_bandwidth python repro.py                                                                                                                        kernel1 run in 0.196224 ms, achieved: 2736.01 GB/s
kernel1 run in 0.190048 ms, achieved: 2824.92 GB/s
kernel1 run in 0.186752 ms, achieved: 2874.78 GB/s
kernel1 run in 0.18512 ms, achieved: 2900.12 GB/s
kernel1 run in 0.185504 ms, achieved: 2894.12 GB/s
```

**Comment by naoyam** [comment]
Closing as the gap was filled
